File: app/src/main/java/com/hqq/example/ui/view/TextViewActivity.java
package com.hqq.example.ui.view;

import android.content.Context;
import android.content.Intent;
import android.text.method.LinkMovementMethod;
import android.view.View;
import android.widget.TextView;

import com.hqq.core.ui.BaseActivity;
import com.hqq.core.utils.ResourcesUtils;
import com.hqq.core.utils.TextSpannableBuilder;
import com.hqq.core.utils.ToastUtils;
import com.hqq.example.R;


/**
 * @Author : huangqiqiang
 * @Package : com.hqq.example.ui.view
 * @FileName :   TextViewActivity
 * @Date : 2020/6/9 0009  下午 3:34
 * @Email :  qiqiang213@gmail.com
 * @Descrive :
 */
public class TextViewActivity extends BaseActivity {
    TextView mTextView8;
    TextView mTextView9;
    TextView mTextView10;

    public static void open(Context context) {
        Intent starter = new Intent(context, TextViewActivity.class);
        context.startActivity(starter);
    }

    @Override
    public int getLayoutViewId() {
        return R.layout.activity_text_view;
    }

    @Override
    public void initView() {
        mTextView10=findViewById(R.id.textView10);
        mTextView9=findViewById(R.id.textView9);
        mTextView8=findViewById(R.id.textView8);
        mTextView10.setMovementMethod(LinkMovementMethod.getInstance());
        mTextView10.setHighlightColor(ResourcesUtils.getColor(R.color.transparent));
        mTextView10.setText(new TextSpannableBuilder()
                .addTextPart("12312312312312312")
                .addTextPart("sddsfgsdfgsdfsgd", ResourcesUtils.getColor(R.color.color_main), new TextSpannableBuilder.OnClickListener() {

                    @Override
                    public void onClick(View widget, CharSequence clickedText) {
                        ToastUtils.showToast("sddsfgsdfgsdfsgd");
                    }
                })
                .addTextPart("12312312312312312")
                .addTextPart("12312312312312312")
                .build());


    }


}

File: app/src/main/java/com/hqq/example/ui/view/TextViewActivity.kt
package com.hqq.example.ui.view

import android.text.method.LinkMovementMethod
import android.text.style.BackgroundColorSpan
import android.text.style.CharacterStyle
import android.text.style.ForegroundColorSpan
import android.view.View
import androidx.compose.ui.graphics.Color
import com.hqq.core.ui.base.BaseViewBindingActivity
import com.hqq.core.utils.ResourcesUtils
import com.hqq.core.utils.ToastUtils
import com.hqq.core.utils.text.TextSpannableBuilder
import com.hqq.example.R
import com.hqq.example.databinding.ActivityTextViewBinding
import com.hqq.example.utils.ColorUtil

/**
 * @Author : huangqiqiang
 * @Package : com.hqq.example.ui.view
 * @FileName :   TextViewActivity
 * @Date : 2020/6/9 0009  下午 3:34
 * @Email :  qiqiang213@gmail.com
 * @Descrive :
 */
class TextViewActivity : BaseViewBindingActivity<ActivityTextViewBinding>() {

    var _textSpannable = TextSpannableBuilder()


    override fun initView() {
        binding.textView10.text = _textSpannable.build()

        binding.button89.setOnClickListener {
            _textSpannable.addTextPart("添加文字")
            binding.textView10.text = _textSpannable.build()
        }

        binding.button90.setOnClickListener {
            _textSpannable.addTextPart("添加文字颜色", ResourcesUtils.getColor(R.color.color_main))
            binding.textView10.text = _textSpannable.build()
        }

        binding.button91.setOnClickListener {
            _textSpannable.addTextPart("添加文字颜色大小", ResourcesUtils.getColor(R.color.color_red), 1.2f, BackgroundColorSpan(ResourcesUtils.getColor(R.color.color_main)), object :
                TextSpannableBuilder.OnClickListener {
                override fun onClick(widget: View?, clickedText: CharSequence?) {
                    ToastUtils.showToast("12312312")
                }

            })
            binding.button91.highlightColor = ResourcesUtils.getColor(R.color.color_757575)
//            _textSpannable.addTextClick("123", ResourcesUtils.getColor(R.color.black),object  :
//                TextSpannableBuilder.OnClickListener{
//                override fun onClick(widget: View?, clickedText: CharSequence?) {
//                    ToastUtils.showToast("12312312")
//                }
//            }
//            )
            binding.textView10.text = _textSpannable.build()
            binding.textView10.setMovementMethod(LinkMovementMethod.getInstance())
        }



        binding.button92.setOnClickListener {

            _textSpannable.addTextPart("添加文字颜色大小", ResourcesUtils.getColor(R.color.color_red), 1.2f, null, object :
                TextSpannableBuilder.OnClickListener {
                override fun onClick(widget: View?, clickedText: CharSequence?) {
                    ToastUtils.showToast("12312312")
                }
            })
//            binding.button91.highlightColor = ResourcesUtils.getColor(R.color.color_757575)
            binding.textView10.setHighlightColor(ResourcesUtils.getColor(R.color.transparent))
            binding.textView10.text = _textSpannable.build()
            binding.textView10.setMovementMethod(LinkMovementMethod.getInstance())

        }

    }


}